code/log/lib/Loot/Log/Syslog.hs

```diff
@@ -0,0 +1,299 @@
+        <*> v .:? "unit-name"   .!= unitName defaultLoggerConfig
+        <*> v .:? "out-options" .!= outOptions defaultLoggerConfig
+        <*> v .:? "facility"    .!= facility defaultLoggerConfig
+        <*> v .:? "min-level"   .!= minLevel defaultLoggerConfig
+        <*> v .:? "sub-loggers" .!= subLoggers defaultLoggerConfig
+    -- | Special parsing for a list of 'SysloggerConfig': this allows the key for
+    -- a 'SysloggerConfig' Object to be used as the "name" key of that same object
+    parseJSONList = withObject "SysloggerConfigList" $
+        mapM (\(name, val) -> parseKeyAsName name val) . H.toList
+      where
+        parseKeyAsName :: Text -> Value -> Parser SysloggerConfig
+        parseKeyAsName k = withObject "SysloggerConfig" $
+            parseJSON . Object . H.insert "name" (String k)
+
+instance ToJSON SysloggerConfig where
+    toJSON SysloggerConfig {..} = object
+        [ "destination" .= destination
+        , "name"        .= loggerName
+        , "unit-name"   .= unitName
+        , "out-options" .= outOptions
+        , "facility"    .= facility
+        , "min-level"   .= minLevel
+        , "sub-loggers" .= subLoggers
+        ]
+    toJSONList = Object . H.fromList . map useNameAsKey
+      where
+        useNameAsKey :: SysloggerConfig -> (Text, Value)
+        useNameAsKey config = case toJSON config of
+            Object obj -> (loggerName config, Object $ H.delete "name" obj)
+            val        -> (loggerName config, val) -- should be impossible
+
+instance FromJSON SyslogDestination where
+    parseJSON = withObject "SyslogDestination" $ \v -> do
+        destType <- v .: "destination-type"
+        case destType :: Text of
+            "auto" -> return AutoLocal
+            "local" -> Local
+                <$> v .:? "fifo-path" .!= "/dev/log"
+            "remote" -> Remote
+                <$> v .:? "family"      .!= AF_INET
+                <*> v .:? "hostname"    .!= "localhost"
+                <*> v .:? "port-number" .!= 514
+            _ -> fail "Parsing SyslogDestination value failed: unknown destination-type"
+
+instance ToJSON SyslogDestination where
+    toJSON = \case
+        AutoLocal -> object
+            [ "destination-type" .= ("auto" :: Text)
+            ]
+        Local fifoPath -> object
+            [ "destination-type" .= ("local" :: Text)
+            , "fifo-path"        .= fifoPath
+            ]
+        Remote family hostname portNum -> object
+            [ "destination-type" .= ("remote" :: Text)
+            , "family"           .= family
+            , "hostname"         .= hostname
+            , "port-number"      .= portNum
+            ]
+
+
+-- Orphan instances:
+instance FromJSON SL.Option where
+    parseJSON = withText "OutOption" $ \case
+        "PID"        -> pure SL.PID
+        "PERROR"     -> pure SL.PERROR
+        "withID"     -> pure SL.PID
+        "withStderr" -> pure SL.PERROR
```

Comment:
Not sure if providing aliases for the same options is actually a good idea. A person who sets up the syslog is likely to be familiar with standard option flag names (`PID` and `PERROR`) anyway, and those aliases are not much more descriptive.